{{cookiecutter.project_name}}/notebooks/ModelValidation.py

```diff
@@ -0,0 +1,224 @@
+# Databricks notebook source
+##################################################################################
+# Model Validation Notebook
+##
+# This notebook uses mlflow model validation API to run mode validation after training and registering a model 
+# in model registry, before deploying it to Production stage.
+#
+# It runs as part of CD and by an automated model training job -> validation -> deployment job defined under ``databricks-config``
+#
+# Please finish the two cells with "TODO" comments before enabling the model validation.
+#
+# Parameters:
+#
+# * env (optional): Name of the environment the notebook is run in (staging, or prod). Defaults to "prod".
+#                   You can add environment-specific logic to this notebook based on the value of this parameter,
+#                   e.g. read validation data from different tables or data sources across environments.
+# * run_mode (optional): The model validation run mode. Default to Disabled. Possible values are Disabled, Dry Run, Enabled.
+#                       Disabled : do not run the model validation notebook
+#                       Dry Run  : Run the model validation notebook. Fail the job if there are compling errors. Ignore failed model validation rules.
+#                       Enabled  : Run the model validation notebook. Return success only if all model validation rules are passing.
+#
+#
+#
+# For details on mlflow evalute API, see doc https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate
+# For details and examples about performing model validation, see the Model Validation documentation https://mlflow.org/docs/latest/models.html#model-validation
+#
+##################################################################################
+
+# COMMAND ----------
+
+# MAGIC %load_ext autoreload
+# MAGIC %autoreload 2
+
+# COMMAND ----------
+
+# MAGIC %pip install -r ../requirements.txt
+
+# COMMAND ----------
+
+dbutils.widgets.dropdown("env", "prod", ["staging", "prod"], "Environment Name")
+dbutils.widgets.dropdown("run_mode", "Disabled", ["Disabled", "Dry Run", "Enabled"], "Run Mode")
+
+# COMMAND ----------
+
+import mlflow
+from mlflow.models import MetricThreshold, make_metric
+import pandas as pd
+import os
+import numpy as np
+import tempfile
+import json
+import traceback
+from mlflow.recipes.utils import (
+    get_recipe_config,
+    get_recipe_name,
+    get_recipe_root_path,
+)
+from mlflow.tracking.client import MlflowClient
+client = MlflowClient()
+
+env = dbutils.widgets.get("env")
+_run_mode = dbutils.widgets.get("run_mode")
+if _run_mode.lower() == "disabled":
+    dbutils.notebook.exit(0)
+dry_run = _run_mode.lower() == "dry run"
+
+def get_model_type_from_recipe():
```

Comment:
Can we please have a notebook (or different method here) that doesn't depend on recipes?

Reply:
Is it feasible?

Reply:
Hmm actually, reading the code below, seems like this  notebook works for both mlflow recipes and other cases, right? This method would be used in the mlflow recipes case.

Reply:
Exactly.

Reply:
I think it's fine to depend on Recipes. We aren't likely to remove the feature in the future.

Reply:
This is a pre-defined function that user can call if they are using recipe. If they are not using recipe, they can hardcode the model type value.
The `get_model_type_from_recipe` is used on line 
```
model_type = get_model_type_from_recipe()
```